webpack/webpack5-module-minifier-plugin/src/AsyncImportCompressionPlugin.ts

```diff
@@ -0,0 +1,314 @@
+
+          if (!localImports) {
+            if (!context.module) {
+              context.compilation.errors.push(
+                new WebpackError(
+                  `Unexpected async import ${token} in non-module context ${context.loggingName}`
+                )
+              );
+              return source;
+            }
+
+            localImports = asyncImportMap.get(context.module);
+            if (!localImports) {
+              context.compilation.errors.push(
+                new WebpackError(`Unexpected async import ${token} in module ${context.loggingName}`)
+              );
+              return source;
+            }
+          }
+
+          const localImport: ILocalImportMetadata | undefined = localImports.get(token);
+          if (!localImport) {
+            context.compilation.errors.push(
+              new WebpackError(`Missing metadata for ${token} in module ${context.loggingName}`)
+            );
+            return source;
+          }
+          const { meta, module } = localImport;
+
+          const chunkExpression: string = meta.index < 0 ? JSON.stringify(meta.chunkIds) : `${meta.index}`;
+
+          source.replace(
+            match.index,
+            ASYNC_IMPORT_REGEX.lastIndex - 1,
+            `${chunkExpression},${JSON.stringify(module.id!)}${getImportTypeExpression(
+              module,
+              context.module!,
+              moduleGraph
+            )}`
+          );
+        }
+
+        return source;
+      });
+
+      compilation.hooks.beforeChunkAssets.tap({ name: PLUGIN_NAME, stage: STAGE_AFTER }, () => {
+        const ImportDependency: typeof WebpackImportDependency = getImportDependency(compilation);
+
+        for (const module of compilation.modules) {
+          let toProcess: Module[];
+
+          if (isConcatenatedModule(module)) {
+            toProcess = module.modules;
+          } else {
+            toProcess = [module];
+          }
+
+          for (const child of toProcess) {
+            processModuleDependenciesRecursive(child, (dep) => {
+              if (dep instanceof ImportDependency) {
+                const targetModule: Module = moduleGraph.getModule(dep);
+
+                if (targetModule) {
+                  let localAsyncImports: Map<string, ILocalImportMetadata> | undefined =
+                    asyncImportMap.get(module);
+                  if (!localAsyncImports) {
+                    asyncImportMap.set(module, (localAsyncImports = new Map()));
+                  }
+
+                  const chunkGroups: ChunkGroup[] = targetModule.blocks.map(chunkGraph.getBlockChunkGroup);
+                  const chunkIds: Set<number | string | null> = new Set();
+
+                  for (const chunkGroup of chunkGroups) {
+                    for (const chunk of chunkGroup.chunks) {
+                      chunkIds.add(chunk.id);
+                    }
+                  }
+
+                  const idString: string = Array.from(chunkIds).join(';');
```

Comment:
```suggestion
                  const idString: string = Array.from(chunkIds).sort().join(';');
```

Reply:
Unless you want to go the bigint bitvector route and use those as order-invariant keys. Might be worth it, but would have to profile and see how much time is spent in this function.